src/core/main.c

```diff
@@ -2449,6 +2457,14 @@ static void reset_arguments(void) {
         arg_random_seed_size = 0;
         arg_clock_usec = 0;
 
+        arg_mount_rate_limit_burst = DEFAULT_MOUNT_RATE_LIMIT_BURST;
+        const char *e = secure_getenv("SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST");
+        if (e) {
+                r = safe_atou(e, &arg_mount_rate_limit_burst);
+                if (r < 0)
+                        log_debug("Invalid value in $SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST, ignoring: %s", e);
+        }
```

Comment:
quite frankly, I'd drop this all here. Just read the env var directly in mount_enumerate(), that's the only place you need the variable, and this is not performance sensitive, given that the function is only called during initialization once. It becomes vastly simpler then and a very local to where the value is consumed.

Reply:
ok, done